.travis.yml

```diff
@@ -1,63 +1,47 @@
 language: python
-sudo: required
-# if the https://travis-ci.org/ODM2/ODM2PythonAPI/requests ever says: missing config
-# validate at: http://lint.travis-ci.org/
-python:
-#  - "2.6"
-  - "2.7"
-#  - "3.2"
-#  - "3.3"
-#  - "3.4"
-#  - "3.5"
-#  - "3.5-dev" # 3.5 development branch
-#  - "nightly" # currently points to 3.6-dev
-# command to install dependencies
-cache:
-  directories:
-    - $HOME/.cache/pip/wheels
-    - $HOME/virtualenv/python2.7.9
-addons:
-  apt:
-    packages:
-      - python-lxml
-      - cmake
-      - unixodbc
-      - unixodbc-dev
-      - odbcinst1debian2
-      - odbcinst
-      - freetds-dev
-      - freetds-bin
-      - tdsodbc
-      - libc6
-      - e2fsprogs
-      - mysql-client
-      - libproj-dev
-      - libgeos-dev
-      - libspatialite-dev
-#services:
- # - mysql
-#  - postgresql
-before_script:
-  - python examples/flask/cbi/build_cbi_cache.py
-  - python examples/flask/swis/runserver_swis.py --config=./test/test_swis_config2.cfg --connection=sqlite:///./test/test_swis2.db &
+
+sudo: false
+
+matrix:
+  fast_finish: true
+  include:
+  - python: 2.7
+    env: TEST_TARGET=default
+  - python: 3.5
+    env: TEST_TARGET=default
+  - python: 3.6
+    env: TEST_TARGET=default
+  - python: 3.6
+    env: TEST_TARGET=coding_standards
+  allow_failures:
+  - python: 3.5
+    env: TEST_TARGET=default
+  - python: 3.6
+    env: TEST_TARGET=default
+  - python: 3.6
+    env: TEST_TARGET=coding_standards
 
 before_install:
-# python -m pip makes the install go into the virtualenv
-  - python -m pip install sqlalchemy requests suds lxml pyodbc
-  - pip install git+https://github.com/ODM2/geoalchemy.git@odm2#egg=geoalchemy-0.7.3
-  - pip install  git+https://github.com/ODM2/ODM2PythonAPI.git@pre_alpha_0#egg=odm2api-0.1.0
+    - wget http://bit.ly/miniconda -O miniconda.sh
+    - bash miniconda.sh -b -p $HOME/miniconda
+    - export PATH="$HOME/miniconda/bin:$PATH"
+    - conda update conda --yes
+    - conda config --add channels odm2 --force
+    - conda create --yes -n TEST python=$TRAVIS_PYTHON_VERSION --file requirements.txt --file requirements-dev.txt
+    - source activate TEST
 
-install: # now just our code
-  - pip install   .
-  # want to be able to test installing dependecies, but
-  # pip install .[odm1] fails (so .[odm1,odm2] fails also)
-#  - pip install pysqlite sqlalchemy requests suds
-  - pip list
+# Test source distribution.
+install:
+    - python setup.py sdist && version=$(python setup.py --version) && pushd dist  && pip install WOFpy-${version}.tar.gz && popd
 
-# don't forget to open up the azure mssql server to these addreses
-# https://docs.travis-ci.com/user/ip-addresses/
+before_script:
+  - python examples/flask/cbi/build_cbi_cache.py
+  - python examples/flask/swis/runserver_swis.py --config=./test/test_swis_config2.cfg --connection=sqlite:///./test/test_swis2.db &
```

Comment:
I though that these would require `mysql-connector-python` and `freetds`, but they are commented out in the `requirements-dev.txt`and the tests passed anyways. I will remove them in my next commit. (I am not familiar with `sql` databases.)